files/image_config/process-reboot-cause/process-reboot-cause

```diff
@@ -70,7 +70,7 @@ def main():
     # if there is no sonic_platform package installed, we only provide
     # software-related reboot causes.
     try:
-        import sonic_platform
+        import sonic_platform.platform
```

Comment:
Why is this necessary?

Reply:
Hi Joe,

sonic_platform is directory. platform is the file.

When we import directory alone We get the following error.
**From File :- /usr/bin/process-reboot-cause**
    import sonic_platform
    # Check if the previous reboot was caused by hardware
    platform = sonic_platform.platform.Platform()
    chassis = platform.get_chassis()
    hardware_reboot_cause, optional_details = chassis.get_reboot_cause()

root@sonic:/home/admin# /usr/bin/process-reboot-cause
Traceback (most recent call last):
  File "/usr/bin/process-reboot-cause", line 121, in <module>
    main()
  File "/usr/bin/process-reboot-cause", line 76, in main
    platform = sonic_platform.platform.Platform()
**AttributeError: 'module' object has no attribute 'platform'**


When we import the file "platform" the script is working
    **import sonic_platform.platform**
    # Check if the previous reboot was caused by hardware
    platform = sonic_platform.platform.Platform()
    chassis = platform.get_chassis()
    hardware_reboot_cause, optional_details = chassis.get_reboot_cause()

root@sonic:/home/admin# /usr/bin/process-reboot-cause
root@sonic:/home/admin#

Please let me if my understanding is fine.

Thanks
Sridhar.R

Reply:
Hmmm. I don't believe this should be necessary. You should be able to import the entire package by specifying the package (directory) name. Let me look into this further.

Reply:
It appears for this to work, you need to modify the __init__.py file of your sonic_platform package to either import the files and/or list the module names in `__all__`. See: https://docs.python.org/3/tutorial/modules.html#importing-from-a-package

Reply:
Hi Joe, 
After some literature study, i was able to successfully make it work. Updated __init__.py with 
__all__ = ["platform", "chassis"]
from sonic_platform import *

On adding these 2 lines, "import sonic_platform" works fine.

On adding new files like sfp.py , psu.py, device,py, we have to update the __init__.py file as well.

Thanks for your help and support Joe.

Thanks
Sridhar.R